File: benchmarks/tsb/bench_expanding_count.ts
/**
 * Benchmark: Expanding.count on 100k-element Series
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Float64Array.from({ length: ROWS }, (_, i) => (i % 10 === 0 ? NaN : Math.sin(i * 0.01)));
const s = new Series(data);

for (let i = 0; i < WARMUP; i++) {
  s.expanding().count();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.expanding().count();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "expanding_count",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_expanding_count.py
"""Benchmark: Expanding.count on 100k-element Series"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = np.where(np.arange(ROWS) % 10 == 0, np.nan, np.sin(np.arange(ROWS) * 0.01))
s = pd.Series(data)

for _ in range(WARMUP):
    s.expanding().count()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.expanding().count()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "expanding_count",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
